Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/english/extractors/EnglishTimePeriodExtractorConfiguration.java

```diff
@@ -0,0 +1,111 @@
+    public static final Pattern TimeUnitRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.TimeUnitRegex);
+    public static final Pattern TimeOfDayRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.TimeOfDayRegex);
+    public static final Pattern PrepositionRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.PrepositionRegex);
+    public static final Pattern TimeFollowedUnit = RegExpUtility.getSafeRegExp(EnglishDateTime.TimeFollowedUnit);
+    public static final Pattern PureNumBetweenAnd = RegExpUtility.getSafeRegExp(EnglishDateTime.PureNumBetweenAnd);
+    public static final Pattern GeneralEndingRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.GeneralEndingRegex);
+    public static final Pattern PeriodHourNumRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.PeriodHourNumRegex);
+    public static final Pattern SpecificTimeFromTo = RegExpUtility.getSafeRegExp(EnglishDateTime.SpecificTimeFromTo);
+    public static final Pattern SpecificTimeBetweenAnd = RegExpUtility.getSafeRegExp(EnglishDateTime.SpecificTimeBetweenAnd);
+    public static final Pattern SpecificTimeOfDayRegex = RegExpUtility.getSafeRegExp(EnglishDateTime.SpecificTimeOfDayRegex);
+    public static final Pattern TimeNumberCombinedWithUnit = RegExpUtility.getSafeRegExp(EnglishDateTime.TimeNumberCombinedWithUnit);
+
+    public EnglishTimePeriodExtractorConfiguration()
+    {
+        this(DateTimeOptions.None);
+    }
+
+    //C# TO JAVA CONVERTER WARNING: The following constructor is declared outside of its associated class:
+    //ORIGINAL LINE: public EnglishTimePeriodExtractorConfiguration(DateTimeOptions options = DateTimeOptions.None)
+    //C# TO JAVA CONVERTER NOTE: Java does not support optional parameters. Overloaded method(s) are created above:
+    public EnglishTimePeriodExtractorConfiguration(DateTimeOptions options)
+    {
+        super(options);
+        TokenBeforeDate = EnglishDateTime.TokenBeforeDate;
+        SingleTimeExtractor = new BaseTimeExtractor(new EnglishTimeExtractorConfiguration(options));
+        UtilityConfiguration = new EnglishDatetimeUtilityConfiguration();
+        IntegerExtractor = com.microsoft.recognizers.text.number.english.extractors.IntegerExtractor.getInstance();
+    }
+
+    private IDateTimeUtilityConfiguration UtilityConfiguration;
+    public final IDateTimeUtilityConfiguration getUtilityConfiguration() { return UtilityConfiguration; }
+
+    private IDateTimeExtractor SingleTimeExtractor;
+    public final IDateTimeExtractor getSingleTimeExtractor() { return SingleTimeExtractor; }
+
+    private IExtractor IntegerExtractor;
+    public final IExtractor getIntegerExtractor() { return IntegerExtractor; }
+
+    public Iterable<Pattern> getSimpleCasesRegex() {
+        return getSimpleCasesRegex;
+    }
+    public final Iterable<Pattern> getSimpleCasesRegex = new ArrayList<Pattern>() {
+        {add(PureNumFromTo);
```

Comment:
{?

Reply:
Just fixed the code format here. This is the `ArrayList` initialization code.